packages/framework/aqueduct/src/data-objects/treeDataObject.ts

```diff
@@ -95,15 +97,15 @@ export abstract class TreeDataObject<TTreeView> extends PureDataObject {
 					`Content with id ${channel.id} is not a SharedTree and cannot be loaded with treeDataObject.`,
 				);
 			}
-			const sharedTree: ITree = channel;
+			const sharedTree: ITree = channel as unknown as ITree;
```

Comment:
it seems like this cast shouldn't be necessary with the above SharedTree.is(channel)? is that not a typeguard? it seems like it should be

Reply:
Yeah, you're right. I actually removed this on another branch. I have another PR I'm working on - I'll make sure this gets included.

Reply:
https://github.com/microsoft/FluidFramework/pull/25013